rust/private/utils.bzl

```diff
@@ -123,3 +123,34 @@ def _get_preferred_artifact(library_to_link):
         library_to_link.interface_library or
         library_to_link.dynamic_library
     )
+
+def expand_locations(ctx):
```

Comment:
@illicitonion @dae could you add some suggestions here and for `_expand_location` to help me fill in the documentation?